optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/header/HeaderView.java

```diff
@@ -37,4 +40,19 @@
     @Inject
     @DataField("tenant-selector")
     private TenantSelectorView tenantSelectorView;
+
+    @Inject
+    @DataField("container")
+    private HTMLDivElement container;
+
+    public void addStickyElement(org.jboss.errai.common.client.api.elemental2.IsElement element) {
+        container.appendChild(element.getElement());
+    }
+
+    public void removeStickyElements() {
+        while (container.childElementCount > 1) {
+            container.lastElementChild.remove();
+        }
```

Comment:
You might want to check `Elemental2DomUtil` out. It has an utility method called `removeAllElementChildren` that does exactly that.

Reply:
`container.childElementCount > 1`; remove all but the first child (which is the menu)

Reply:
Oops, okay. Maybe it's better to explicitly say you're not deleting the menu by filtering or writing an if statement inside the while body.. The position of the DOM element is too loosely coupled with this method.